static/js/chart/draw.ts

```diff
@@ -88,6 +99,200 @@ function appendLegendElem(
     .attr("href", (d) => d.link || null);
 }
 
+/**
+ * Adds tooltip element within a given container.
+ *
+ * @param containerId
+ */
+function addTooltip(containerId: string): void {
+  d3.select("#" + containerId)
+    .attr("style", "position: relative")
+    .append("div")
+    .attr("id", TOOLTIP_ID)
+    .attr("style", "position: absolute; display: none; z-index: 1");
+}
+
+/**
+ * Position and show the tooltip.
+ *
+ * @param contentHTML innerHTML of the tooltip as a string
+ * @param containerId
+ * @param datapointX
```

Comment:
please describe the other fields as well

Reply:
done